### PR TITLE
[CostModel] Mark ssa_copy as free

### PR DESCRIPTION
**Pull request by davemgreen**
These are intrinsics are only used ephemerally and be should be given a zero cost.

### CODE REVIEW

**Comment by fhahn** [comment]
I am curious if you encountered cases where those intrinsics are still around after the middle-end optimization pipeline. I think all users of those intrinsics should clean them up, but there might be some places that got missed.

**Comment by davemgreen** [comment]
No the costing is done during a pass that adds and removes them. This could be handled specially in that pass, but it is more general fix to give a zero cost to everywhere.

File: llvm/include/llvm/CodeGen/BasicTTIImpl.h [hunk 1]
```diff
@@ -1510,6 +1510,8 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
     Intrinsic::ID IID = ICA.getID();
     if (Function::isTargetIntrinsic(IID))
       return TargetTransformInfo::TCC_Basic;
+    if (IID == Intrinsic::ssa_copy)
+      return TargetTransformInfo::TCC_Free;
```
Comment (nikic) [inline llvm/include/llvm/CodeGen/BasicTTIImpl.h:1514]: This should be in TargetTransformInfoImpl.h.

Comment (davemgreen) [reply llvm/include/llvm/CodeGen/BasicTTIImpl.h]: Ah I see. That is better, thanks.

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**